src/Core/Manifest/Module.php

```diff
@@ -152,63 +181,69 @@ protected function loadComposer()
     }
 
     /**
-     * Gets path to physical file resource relative to base directory.
-     * Directories included
+     * Get resource for this module
      *
-     * This method makes no distinction between public / local resources,
-     * which may change in the near future.
-     *
-     * @internal Experimental API and may change
-     * @param string $path File or directory path relative to module directory
-     * @return string Path relative to base directory
+     * @param string $path
+     * @return ModuleResource
      */
-    public function getRelativeResourcePath($path)
+    public function getResource($path)
     {
-        $base = trim($this->getRelativePath(), '/\\');
         $path = trim($path, '/\\');
-        return trim("{$base}/{$path}", '/\\');
+        if (isset($this->resources[$path])) {
+            return $this->resources[$path];
+        }
+        return $this->resources[$path] = new ModuleResource($this, $path);
     }
 
     /**
-     * Gets path to physical file resource relative to base directory.
-     * Directories included
-     *
-     * This method makes no distinction between public / local resources,
-     * which may change in the near future.
-     *
-     * @internal Experimental API and may change
-     * @param string $path File or directory path relative to module directory
-     * @return string Path relative to base directory
+     * @deprecated 4.0...5.0 Use getResource($path)->getRelativePath() instead
+     * @param string $path
+     * @return string
+     */
+    public function getRelativeResourcePath($path)
```

Comment:
Do we need to change other core modules to not use this deprecated API? :D

Reply:
Gah.

Reply:
We could but it's not critical; We should be forgiving if people treat previous betas as "final api";

Reply:
Works for me, as long as you track that remaining work against the 4.0 stable milestone as a new ticket :)